### PR TITLE
Ensure the generated changelog gets linted, fix backticks in log

### LINKED ISSUE #1189: Release drafter removes words inside back tick quotes when creating a PR
## Description

The [PR](https://github.com/WordPress/openverse/pull/1185/files) that would publish the API changelog removes all the words that were in back tick quotes. The [GitHub version](https://github.com/WordPress/openverse/releases/tag/api-2023.04.12.23.29.59), interestingly, is not affected.

## Reproduction

See, for example, line 15 here: https://github.com/WordPress/openverse/pull/1186/files

## Screenshots

`- Absorb `build-nginx` job into `build-images` job (#944) @dhruvkb` becomes `- Absorb job into job (#944) @dhruvkb`

## Additional context

It seems that we could use [replacers](https://github.com/release-drafter/release-drafter#replacers) to fix this issue.
One more thing that we might want to fix when publishing the changelogs to docs.openverse.org would be the links to the PRs. GitHub automatically inserts them when it sees the PR numbers, but our docs site won't.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #1189 by @obulat, also fixes an issue preventing the linting from happening on changelog PRs like #1186

## Description

This PR modifies the `release-app` workflow in two ways:

1. It stages the generated changelog in git to _specifically ensure_ that `just lint` runs on it. For whatever reason, leaving it as an unstaged file prevents `just lint` from running on it, even though I cannot reproduce that behavior locally (i.e. unstaged & untracked files _are_ linted locally).
2. It changes the way the body of the changelog is inserted into a file so that backticks (`) don't get interpolated by bash. I've confirmed that using single quotes should prevent that locally, but I can't test this fully yet until we have a release that includes backticks in the file.

**Successful release run**: https://github.com/WordPress/openverse/actions/runs/4694770892/jobs/8323238328
**Already-linted changelog PR**: https://github.com/WordPress/openverse/pull/1192

## Testing Instructions

View the above!

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .github/workflows/release-app.yml [hunk 1]
```diff
@@ -110,11 +110,9 @@ jobs:
...
       - name: Add new changelog file to documentation
         run: |
-          cat << EOF > documentation/changelogs/${{ inputs.app }}/${{ steps.tag.outputs.date }}.md
-          # ${{ steps.tag.outputs.date }}
-
-          ${{ steps.release-drafter.outputs.body }}
-          EOF
+          file="documentation/changelogs/${{ inputs.app }}/${{ steps.tag.outputs.date }}.md"
+          echo -e "# ${{ steps.tag.outputs.date }}\n\n" > "$file"
+          echo -e '${{ steps.release-drafter.outputs.body }}' >> "$file"
```
Comment (obulat) [inline .github/workflows/release-app.yml:115]: Was it bash here that removed the backticks? 🤯

Comment (AetherUnbound) [reply .github/workflows/release-app.yml:115]: Yep! I mention this in the linked issue 😄 https://github.com/WordPress/openverse/issues/1189#issuecomment-1507613278

**Comment by zackkrida** [comment]
@AetherUnbound, @obulat and @dhruvkb I'm upping the priority to high here; I think it's quite important that we're able to release reliably and easily. I'll review this myself now as well.

**Review by zackkrida — APPROVED** [review]
The script changes to prevent interpreting backticks as command substitutions looks correct, nice work!

**Review by obulat — APPROVED** [review]
Nice! Thank you for explanation :)

**Pull request merged**